lib/agents/reference_matcher.py

```diff
@@ -0,0 +1,61 @@
+"""Agent for matching bibliographic references to documents."""
+
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_core.runnables.config import RunnableConfig
+from pydantic import BaseModel, Field
+
+from lib.config.llm_models import gpt_5_mini_model
+from lib.models.agent import LangChainAgent
+
+
+class ReferenceMatchResult(BaseModel):
+    """Result of matching a reference to documents."""
+
+    matched_index: int = Field(
+        description="1-based index of matched document, or -1 if no match"
+    )
+    confidence: str = Field(
+        description="Confidence level: high, medium, low, or none"
+    )
+    reasoning: str = Field(description="Brief explanation for the match decision")
+
+
+_reference_matcher_prompt = ChatPromptTemplate.from_template(
+    """You are matching bibliographic references to documents. Given a reference citation text, determine which document (if any) it cites.
+
+## Reference to match:
+"{reference_text}"
+
+## Available documents:
+{candidates}
+
```

Comment:
instead of passing a single reference item and multiple candidate documents, have you considered passing a single candidate document and the full list of reference items? my initial thought is that this would scale better, because fitting multiple candidate documents could blow up the context pretty fast, while the full list of reference items will not grow as fast. That way we might not even need to pass only the summaries of the candidate supporting files, but the full supporting file (or truncanting to the first X characters)

Reply:
@CarlosBonetti, I've implemented this one for now, and it worked well.
LMK if you have any suggestions or thoughts about it.